src/Cardano/Signer.cpp

```diff
@@ -209,28 +205,89 @@ vector<TxInput> Signer::selectInputsSimple(const vector<TxInput>& inputs, Amount
     return selected;
 }
 
+// Select a subset of inputs, to cover desired token amount. Simple algorithm: pick largest ones.
+void selectInputsSimpleToken(const vector<TxInput>& inputs, string key, uint256_t amount, vector<TxInput>& selectedInputs) {
+    uint256_t selectedAmount = 0;
+    for (const auto& si: selectedInputs) {
+        selectedAmount += si.tokenBundle.getAmount(key);
+    }
+    if (selectedAmount >= amount) {
+        return; // already covered
+    }
+    auto ii = vector<TxInput>(inputs);
+    sort(ii.begin(), ii.end(), [key](TxInput t1, TxInput t2) {
+        return t1.tokenBundle.getAmount(key) > t2.tokenBundle.getAmount(key);
+    });
+    for (const auto& i: ii) {
+        if (distance(selectedInputs.begin(), find(selectedInputs.begin(), selectedInputs.end(), i)) < selectedInputs.size()) {
+            // already selected
+            continue;
+        }
+        selectedInputs.push_back(i);
+        selectedAmount += i.amount;
+        if (selectedAmount >= amount) {
+            return;
+        }
+    }
```

Comment:
I'm almost sure that https://en.cppreference.com/w/cpp/algorithm/nth_element can result as the same logic for this code part. (from what I understand you want to pick the largest ones)

Reply:
The accumulate suggestion I accepted.
The bigger for cycle can't be simplified much I'm afraid; a subset of the inputs has to be selected that covers the minimum sum across possible multiple tokens.  The token amounts differ in the inputs.  Selection has to take into account that some inputs may already be selected due to previous tokens.

Reply:
> The accumulate suggestion I accepted. The bigger for cycle can't be simplified much I'm afraid; a subset of the inputs has to be selected that covers the minimum sum across possible multiple tokens. The token amounts differ in the inputs. Selection has to take into account that some inputs may already be selected due to previous tokens.

Thanks for the update! Looks better now!